docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/src/Plugin/Validation/Constraint/ValidatePrettyUrlConstraint.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Drupal\cgov_site_section\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * A composite constraint validating Site Section pretty url.
+ *
+ * @Constraint(
+ *   id = "CgovValidatePrettyUrlConstraint",
+ *   label = @Translation("ValidatePrettyUrlConstraint", context="Validation"),
+ * )
+ */
+class ValidatePrettyUrlConstraint extends Constraint {
+
+  /**
+   * Unique pretty url.
+   *
+   * @var string
+   */
+  public $uniquePrettyUrl = 'The site section cannot be saved. Site sections must have unique pretty url.';
```

Comment:
@andyvanavery31 and @sarinapadilla - would one of you please chime in with text for the error message?  [The original ticket](https://github.com/NCIOCPL/cgov-digital-platform/issues/175) only specifies that they must be unique.

Reply:
@navsunka @blairlearn updated the ticket #3170 with text for the error message in the Expected Results section.